locust/templates/index.html

```diff
@@ -67,17 +67,17 @@ <h2>Start new load test</h2>
                         <input type="hidden" name="spawn_rate" id="spawn_rate" value="0"/><br>
                     {% else %}
                         <label for="user_count">Number of total users to simulate</label>
-                        <input type="text" name="user_count" id="user_count" class="val" value="{{ num_users or "" }}"/><br>
+                        <input placeholder="Please enter a number" type="number" required="required" name="user_count" id="user_count" class="val" value="{{ num_users or "" }}"/><br>
                         <label for="spawn_rate">Spawn rate <span style="color:#8a8a8a;">(users spawned/second)</span></label>
-                        <input type="text" name="spawn_rate" id="spawn_rate" class="val" value="{{ spawn_rate or "" }}"/><br>
+                        <input placeholder="Please enter a number" type="number" required="required" name="spawn_rate" id="spawn_rate" class="val" value="{{ spawn_rate or "" }}"/><br>
                     {% endif %}
                     <label for="host">
                         Host <span style="color:#8a8a8a;">(e.g. http://www.example.com)</span>
                         {% if override_host_warning %}
                             <span style="color:#8a8a8a; font-size:12px;">(setting this will override the host on all User classes)</span></label>
                         {% endif %}
                     </label>
-                    <input type="text" name="host" id="host" class="val" autocapitalize="off" autocorrect="off" value="{{ host  or "" }}"/><br>
+                    <input placeholder="Please enter a Host" type="text" required="required" name="host" id="host" class="val" autocapitalize="off" autocorrect="off" value="{{ host  or "" }}"/><br>
```

Comment:
The host shouldn't be mandatory.

Reply:
If the host is not mandatory means that all the test client (HTTP) requests need to be using absolute URLs, not relative ones. I'm ok with that but there are some checks inplace to enforce the value for host. 
Here is the log that throws FastHttpUser and HttpUser:

https://github.com/locustio/locust/blob/db2f7649d32191beb359d0a6f9c86342e63d545e/locust/contrib/fasthttp.py#L310-L313
```log
  File "/Users/axel/git-workspace/locust/locust/contrib/fasthttp.py", line 312, in __init__
    "You must specify the base host. Either in the host attribute in the User class, or on the command line using the --host option."
locust.exception.LocustError: You must specify the base host. Either in the host attribute in the User class, or on the command line using the --host option.
2020-12-16T15:29:17Z <Greenlet at 0x7fe7effda290: <lambda>> failed with LocustError
```
https://github.com/locustio/locust/blob/db2f7649d32191beb359d0a6f9c86342e63d545e/locust/user/users.py#L212-L215
```log
  File "/Users/axel/git-workspace/locust/locust/user/users.py", line 214, in __init__
    "You must specify the base host. Either in the host attribute in the User class, or on the command line using the --host option."
locust.exception.LocustError: You must specify the base host. Either in the host attribute in the User class, or on the command line using the --host option.
```

Reply:
I think that host it's not exactly needed in order to make client requests working. It's better to check if the URL is relative and there is no base_url(host value) in the session to build the full URL
Let me know and I could make the adjustments and remove the host validations Locust errors

Reply:
People might be using custom `User` classes that doesn't use the HTTP client. For `HttpUser` and `FastHttpUser` those checks makes sense, but the Web UI will be the same even if you're using a custom `User` class that doesn't use the host.

Reply:
That's not completely True, host it's just needed if the requests doesn't use an absolute URL. Otherwise the host value wouldn't been used at all. I think it should concern of the Http User Test Class to provide the value of the host for base URL or not. Also it could be the case that 1 test class use a host value and others Test Classes use a different host value. I don't see the point of asking for a host value in the form before swarm begin when maybe it wasn't been used by any User Test Class. 
I think it's more consistent to have a base host value in the argparse options that all the Test Classes could pick if they don't provide one. Because in the form you couldn't pick which User Test Classes will get the value you supply for their host field or provide a value for each of the User Test Classes that will be running when the swarm begins

Reply:
Yes, one could argue for removing the host input from the UI altogether. However, people *are* using that functionality in their Locust work flows, and removing it for those user is in my opinion much worse than the negative effect of showing an empty input for people not using it.

In any case, I think it's better to leave that out of the scope of this PR.